core/src/protocols/stream/quic/crypto.rs

```diff
@@ -0,0 +1,295 @@
+// This is heavily based on Cloudflare's Rust implementation of QUIC, known as Quiche.
```

Comment:
similar to general comment -- at some point, header comment would be helpful for us (I'm only surface-level familiar with inner workings of QUIC)

Reply:
Will do!